packages/runtime/src/components/page/PageHead.tsx

```diff
@@ -0,0 +1,294 @@
+            data-makeswift-meta="true"
+          />
+          <meta name="twitter:image" content={socialImage.publicUrl} data-makeswift-meta="true" />
+          <meta name="twitter:card" content="summary_large_image" data-makeswift-meta="true" />
+          <meta itemProp="image" content={socialImage.publicUrl} data-makeswift-meta="true" />
+        </>
+      )}
+
+      {fontFamilyParamValue !== '' && (
+        <link
+          rel="stylesheet"
+          href={`https://fonts.googleapis.com/css?family=${fontFamilyParamValue}&display=swap`}
+          data-makeswift-meta="true"
+        />
+      )}
+
+      {headSnippets.map(snippetToElement).map(children =>
+        Children.map(children, child => {
+          if (typeof child === 'string') return child
+
+          if ((child.type as string) === 'script') {
+            return cloneElement(child, { ref: saveScriptRef })
+          }
+
+          if (VALID_HEAD_ELEMENT_TYPES.includes(child.type as string)) return child
+
+          return null
+        }),
+      )}
+    </>
+  )
+
+  // Won't inject head tag if in pages router (useServerInsertedHTML is a no-op)
+  // The ref is used to prevent a duplicate insertion:
+  // https://github.com/vercel/next.js/discussions/49354
+  const insertedServerHTML = useRef(false)
+  useServerInsertedHTML(() => {
+    if (insertedServerHTML.current) return
+    insertedServerHTML.current = true
+    return headElementsFragment
+  })
+
+  const isPagesRouter = useIsPagesRouter()
+  const [shouldRenderPortal, setShouldRenderPortal] = useState(false)
+
+  // Purge the initial head elements that were server inserted. Then,
+  // we can portal in the same head elements so they will now be tracked on
+  // client side navigation by React. Will not occur in pages router.
+  useEffect(() => {
+    if (isPagesRouter) return
+    document.head.querySelectorAll('[data-makeswift-meta]').forEach(el => el.remove())
+    setShouldRenderPortal(true)
+  }, [isPagesRouter])
+
+  // Scripts portalled into the head tag won't execute. To make them execute, we
+  // use the script refs to append duplicates nodes to the head.
+  useEffect(() => {
+    if (!shouldRenderPortal) return
+
+    const appendedScripts: HTMLScriptElement[] = []
+
+    async function executeScriptsInOrder() {
+      scriptRefs.current.forEach(async node => {
+        await new Promise<void>(resolve => {
+          if (node == null) return resolve()
+          const script = document.createElement('script')
+          script.setAttribute('temp', 'true')
+          script.textContent = node.textContent
+          Array.from(node.attributes).forEach(({ name, value }) => {
+            script.setAttribute(name, value)
+          })
+
+          script.onload = () => resolve()
+          script.onerror = () => resolve()
+
+          node.parentNode?.appendChild(script)
+          appendedScripts.push(script)
+
+          if (!script.hasAttribute('src')) resolve()
+        })
+      })
```

Comment:
These promises seem to be swallowed and unused. I don't think the top-level `.catch` will work. For it to work this would be to need a `Promise.all` on a `scripts.current.map` where we map all script re-executions into promises. Furthermore, we would want `script.onerror` to be `reject`, not `resolve`.